### PR TITLE
Rules for the week

### LINKED ISSUE #520: linked against C++ standard library
## Summary

Indicative of C++, most likely MS Visual C++.

## Features

- string: `Copyright (c) 1992-2004 by P.J. Plauger, licensed by Dinkumware, Ltd. ALL RIGHTS RESERVED.`
etc.

### Namespace
linking/static/cpp

### References

https://en.wikipedia.org/wiki/P._J._Plauger

### LINKED ISSUE #541: FPs encrypt-data-using-rc4-ksa.yml
## Summary

False positive on RC4 KSA rule

## Examples

- `8333822ed41d9f2b302cf8e21b126efc:0x40646a`

## Possible improvements

- `modulo key length` rule could be inside a basic block that is also a tight loop and also checks for `0x100` and/or `0xFF` instead of checking against the whole function?

### LINKED ISSUE #640: dotnet: access resource at runtime
- https://learn.microsoft.com/en-us/dotnet/api/system.resources.resourcemanager?redirectedfrom=MSDN&view=net-7.0
- https://learn.microsoft.com/en-us/dotnet/api/system.reflection.assembly.getmanifestresourcestream?view=net-7.0

### PR DESCRIPTION
**Pull request by mr-tz**
closes #640
closes #541
closes #520

### CODE REVIEW

File: data-manipulation/encryption/rc4/encrypt-data-using-rc4-ksa.yml [hunk 1]
```diff
@@ -12,9 +12,8 @@ rule:
       - Cryptography::Encryption Key::RC4 KSA [C0028.002]
     examples:
       - 34404A3FB9804977C6AB86CB991FB130:0x403D40
-      - C805528F6844D7CAF5793C025B56F67D:0x4067AE
```
Comment (mr-tz) [inline data-manipulation/encryption/rc4/encrypt-data-using-rc4-ksa.yml:15]: I did not find this sample in testfiles?!

File: data-manipulation/encryption/rc4/encrypt-data-using-rc4-ksa.yml [hunk 2]
```diff
@@ -12,9 +12,8 @@ rule:
...
       - 9324D1A8AE37A36AE560C37448C9705A:0x404950
-      - 782A48821D88060ADF0F7EF3E8759FEE3DDAD49E942DAAD18C5AF8AE0E9EB51E:0x405C42
```
Comment (mr-tz) [inline data-manipulation/encryption/rc4/encrypt-data-using-rc4-ksa.yml:17]: stored using MD5 name in testfiles, updated here

File: linking/static/linked-against-cpp-standard-library.yml [hunk 3]
```diff
@@ -0,0 +1,16 @@
+rule:
+  meta:
+    name: linked against CPP standard library
```
Comment (williballenthin) [inline linking/static/linked-against-cpp-standard-library.yml:3]: should we use "C++" here?

**Review by williballenthin — APPROVED** [review]
nice!

File: linking/static/linked-against-cpp-standard-library.yml (hunk 3 above)
Comment (mr-tz) [reply linking/static/linked-against-cpp-standard-library.yml:3]: All existing rules use CPP and I think we'd have to update the linter.

**Pull request merged**